tests/test_utils.py

```diff
@@ -2251,7 +2251,15 @@ def test_to_segmented_index(obj, allow_nat, files_duration, root, expected):
                 audformat.filewise_index(['f1', 'f2']),
                 audformat.filewise_index(['f2', 'f3']),
             ],
-            audformat.filewise_index('f3'),
+            audformat.filewise_index(['f2', 'f3']),
```

Comment:
Turns out symmetric difference is slightly more complicated and we had a few tests wrong.

Here we have:
```
A = ['f1', 'f2']
B = ['f1', 'f2']
C = ['f2', 'f3']
Then we get:
(A Δ B) Δ C = [] Δ ['f2', 'f3'] = ['f2', 'f3'] = ['f1', 'f2'] Δ ['f1', 'f3'] = A Δ (B Δ C)

Reply:
I would argument that the result is correct since `f3` is the only item that is only in one of the three index objects.

Reply:
I would not apply it in sequence but look at all three objects at once.

Reply:
OK, but this is not the definition of symmetric difference.

Reply:
Maybe we should define it as `union() - interesct()`, then the ouput will be `['f1', 'f3']`.

Reply:
You would do `union() - intersect()` also piecewise and the output would be the same. This was my first approach of implementing it.

Reply:
See https://github.com/audeering/audformat/pull/259/commits/7e099899429d6a26ff72c5812b64e2ecf513ff49

Reply:
If a user wants to have your behavior she can still do:
```python
symmetric_difference([index1, union([index2, index3])])
```

Reply:
But then you are again splitting up the objects. For me the function is seems pretty useless if we define it like this.

Reply:
I would only make sense if you had `obj.symmetric_difference(other_objects)`.

Reply:
Sorry, forget about my last two comments :)

Reply:
So, neither with the old nor with the new implementation `symmetric_difference(objs) == union(objs) - intersect(objs)` holds, correct?

Reply:
So I think we should ask ourselves what is the use-case for the function. If the got with `['f2', 'f3']` (even if this is correct according to the definition of symmetric difference), I don't see at the moment when this will ever be useful.

Reply:
Yes, symmetric difference for three objects would be

![venn-symmetric-difference](https://user-images.githubusercontent.com/173624/182379530-bdcbba54-a731-4da8-ba23-158d52e31bf4.png)

What you want to have is

![venn-desired](https://user-images.githubusercontent.com/173624/182379548-2cf86be5-a102-447f-a797-ba4ba6d4feaf.png)

Maybe, this is indeed better when using it in our cases, but I think we need to find a different name, or make sure to describe it correctly in the docstring.

Reply:
Funny, there seems to be no name for it: https://math.stackexchange.com/questions/2415928/union-of-differences-between-three-sets-where-nothing-intersects-how-is-it-cal :)

An alternative would be that the function takes two objects instead of a list of objects as input. But then it would not fit so well with `union()` and `intersect()`.

Reply:
I would not change it to take only two objects.

I see the following options for switching to the behavior proposed by you:
a) stay with `symmetric_difference()` and the current implementation + add something like `difference()` that has your proposed behavior
b) stay with `symmetric_difference()` and add a comment to the docstring how it behaves for more than two objects
c) change the name to something like `difference()` and change to the proposed behavior

Reply:
I implemented c) in https://github.com/audeering/audformat/pull/264